### PR TITLE
Update README for initial core library release

### PR DESCRIPTION
**Pull request by dazuma**
Fixes https://github.com/census-instrumentation/opencensus-ruby/issues/38

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,41 +34,60 @@ gem "opencensus"
...
+# In config.ru or similar Rack configuration file
+require "opencensus/trace/integrations/rack_middleware"
 use OpenCensus::Trace::Integrations::RackMiddleware
 ```
 
+## Instrumentation features
+
+### Tracing outgoing HTTP requests
+
+If you request uses the [Faraday](https://github.com/lostisland/faraday)
```
Comment (chingor13) [inline README.md]: "you request" -> "your app"?

File: README.md [hunk 2]
```diff
@@ -27,41 +34,60 @@ gem "opencensus"
...
+library to make outgoing HTTP requests, consider installing the Faraday
+Middleware integration. This creates a span for each outgoing Faraday request,
+tracking the latency of that request, and it propagates distributed trace
```
Comment (chingor13) [inline README.md]: I think you can remove the "it"

File: README.md [hunk 3]
```diff
@@ -27,41 +34,60 @@ gem "opencensus"
...
+```ruby
+conn = Faraday.new(url: "http://www.example.com") do |c|
+  c.use OpenCensus::Trace::Integrations::FaradayMiddleware
+  c.adapter Faraday.default_adapter
+  end
+conn.get "/"
+```
+
+See the documentation for the
+`OpenCensus::Trace::Integrations::FaradayMiddleware` class for more info.
```
Comment (chingor13) [inline README.md]: We are generating YARD docs so maybe add a link to http://opencensus.io/opencensus-ruby/api/OpenCensus/Trace/Integrations/FaradayMiddleware.html

File: README.md [hunk 4]
```diff
@@ -73,20 +99,60 @@ OpenCensus::Trace.in_span "my_task" do |span|
 end
 ```
 
+See the documentation for the `OpenCensus::Trace` module for more info.
```
Comment (chingor13) [inline README.md]: Link to API docs. http://opencensus.io/opencensus-ruby/api/OpenCensus/Trace.html

File: README.md (hunk 1 above)
Comment (dazuma) [reply README.md]: fixed

File: README.md (hunk 2 above)
Comment (dazuma) [reply README.md]: ok

File: README.md (hunk 3 above)
Comment (dazuma) [reply README.md]: Done

File: README.md (hunk 4 above)
Comment (dazuma) [reply README.md]: Done

**Comment by dazuma** [comment]
Updated docs/index.md as well.

**Pull request merged**